charts/actions-runner-controller/templates/deployment.yaml

```
@@ -84,7 +84,15 @@ spec:
               name: {{ include "actions-runner-controller.secretName" . }}
               optional: true
         - name: GITHUB_APP_PRIVATE_KEY
+          {{- if .Values.authSecret.github_private_key_from_disk }}
           value: /etc/actions-runner-controller/github_app_private_key
+          {{- else }}
+          valueFrom:
+            secretKeyRef:
+              key: github_app_private_key
+              name: {{ include "actions-runner-controller.secretName" . }}
+              optional: true
+          {{- end }}
```

@apr-1985 @marchenkov @toast-gear Can we just do this instead, without the extra flag?

```suggestion
          valueFrom:
            secretKeyRef:
              key: github_app_private_key
              name: {{ include "actions-runner-controller.secretName" . }}
              optional: true
```

Regardless of we use the local file or the secretKeyRef, the content of the private key is loaded from `controller-manager` secret anyway. Then, this won't be a breaking change, right?

---

@mumoshu, I can confirm that your variant works for me too. My main problem - that file `/etc/actions-runner-controller/github_app_private_key` not found. So both options are valid for me, but your proposal the better. Thanks

---

Thanks for your confirmation! Let's go ahead without the if block then.